index.js

```diff
@@ -122,6 +129,16 @@ function compressPlugin (fastify, opts, next) {
       // don't compress on missing or identity `accept-encoding` header
       ((encoding = getEncodingHeader(encodings, req)) == null || encoding === 'identity')
 
+    if (encoding == null && onUnsupportedEncoding != null) {
+      var encodingHeader = req.headers['accept-encoding']
+      var errorPayload = onUnsupportedEncoding(encodingHeader, reply.request, reply)
+      if (errorPayload instanceof Error) {
+        return next(errorPayload)
+      } else {
+        return next(null, errorPayload)
+      }
```

Comment:
I think returning an Error is not a common pattern in JS (and Fastify). Errors are thrown.

Reply:
Here is the guidance to pass it as a param to next: https://www.fastify.io/docs/latest/Hooks/#manage-errors-from-a-hook

Reply:
Good catch!